repose-aggregator/core/repose-core/src/main/scala/org/openrepose/commons/utils/classloader/EarClassProvider.scala

```diff
@@ -143,23 +142,26 @@ class EarClassProvider(earFile: File, val outputDir: File) {
         if (entry.isDirectory) {
           entryFile.mkdir()
         } else {
-          val ofs = new FileOutputStream(entryFile)
+          val entryFileOut = new FileOutputStream(entryFile, true)
           try {
+            val entryFileChannel = entryFileOut.getChannel
+
             log.trace("Obtaining file lock on: {}", entryFile)
-            val lock = ofs.getChannel.lock()
-            try {
-              if (!entryFile.exists() || entry.getCrc != checksumCRC32(entryFile)) {
-                log.trace("Unpacking: {}", entryFile)
-                Stream.continually(zis.read(buffer)).takeWhile(_ != -1).foreach(count => ofs.write(buffer, 0, count))
-              } else {
-                log.trace("File already exists in a valid condition. Skipping: {}", entryFile)
-              }
-            } finally {
-              log.trace("Releasing file lock on: {}", entryFile)
-              lock.release()
+            entryFileChannel.lock()
+
+            if (entry.getSize != entryFile.length) {
+              log.trace("Unpacking: {}", entryFile)
+              // Clear the current contents of the file
+              entryFileChannel.truncate(0)
+              // Write the zip entry contents to the file
+              Stream.continually(zis.read(buffer)).takeWhile(_ != -1).foreach(count => entryFileOut.write(buffer, 0, count))
```

Comment:
At one point I actually replaced the `truncate` and `write` lines with simply:
```
Files.copy(earInputStream, entryFile.toPath, REPLACE_EXISTING)
```

While that does clean things up a bit, it is not exactly safe or clear to use. `Files.copy` deletes the file on disk before writing a new file. While this does not break locking on my machine, it may on others. Furthermore, writing to the `FileOutputStream` or its `Channel` after calling `Files.copy` will not have any effect on the file that was written by `Files.copy`.